library/Network/CloudSeeder/Main.hs

```diff
@@ -174,54 +189,71 @@ getTags config stackToProvision env appName =
 -- and misconfigurations.
 getParameters
   :: forall e m. (AsCliError e, MonadError e m, MonadCLI m, MonadEnvironment m, MonadCloud m)
-  => S.Set (T.Text, ParameterSource) -- ^ parameter sources to fetch values for
+  => ProvisionType
+  -> S.Set (T.Text, ParameterSource) -- ^ parameter sources to fetch values for
   -> S.Set ParameterSpec -- ^ parameter specs from the template currently being deployed
   -> [T.Text] -- ^ names of stack dependencies
   -> T.Text -- ^ name of environment being deployed to
   -> T.Text -- ^ name of application being deployed
-  -> m (M.Map T.Text T.Text)
-getParameters paramSources paramSpecs dependencies env appName = do
-    let constants = paramSources ^..* folded.aside _Constant
-    fetchedParams <- S.unions <$> sequence [envVars, flags, outputs]
-    let allParams = S.insert ("Env", env) (constants <> fetchedParams)
-    validateParameters allParams
+  -> m (M.Map T.Text (Maybe T.Text))
+getParameters provisionType paramSources allParamSpecs dependencies env appName = do
+    let paramSpecs = setRequiredSpecsWithPreviousValuesToOptional allParamSpecs
+        constants  = S.map (second Just) $ paramSources ^..* folded.aside _Constant
+    fetchedParams  <- S.unions <$> sequence [envVars, flags paramSpecs, outputs]
+    let allParams = S.insert ("Env", Just env) (constants <> fetchedParams)
+    validateParameters paramSpecs allParams
+
   where
-    envVars :: m (S.Set (T.Text, T.Text))
+    setRequiredSpecsWithPreviousValuesToOptional :: S.Set ParameterSpec -> S.Set ParameterSpec
+    setRequiredSpecsWithPreviousValuesToOptional pSpecs = do
+      let previousParamKeys = provisionType ^. _UpdateStack -- get params from extant stack if update
+          requiredParamSpecs :: S.Set T.Text = pSpecs ^..* folded._Required -- get required param specs from template
+          optionalParamSpecs :: S.Set (T.Text, Maybe T.Text) = pSpecs ^..* folded._Optional
```

Comment:
You should probably remove these type signatures unless you think they meaningfully improve readability.